Add `assertThrowsExactly` assertion method (#2643)
junit-team/junit5: documentation/src/docs/asciidoc/release-notes/release-notes-5.8.0-M1.adoc

* `DynamicTests.stream()` can now consume `Named` input and will use each name-value pair
  as the display name and value for each generated dynamic test (see
  <<../user-guide/index.adoc#writing-tests-dynamic-tests-examples,User Guide>> for details).

> * New `assertThrowsExactly` method is introduced as a more strict version of `assertThrows`

Reviewer: This also needs to be moved to `release-notes-5.8.0-M2.adoc`.

Author: Done.

> * New `assertThrowsExactly` method is introduced as a more strict version of `assertThrows`
>   allowing you to assert that the thrown exception has the exact specified class.

Reviewer: ```suggestion
* New `assertThrowsExactly` method which is a more strict version of `assertThrows`
  that allows you to assert that the thrown exception has the exact specified class.
```
